src/data/battle-anims.ts

```diff
@@ -1045,6 +1264,24 @@ export class MoveChargeAnim extends MoveAnim {
   }
 }
 
+export class EncounterBattleAnim extends BattleAnim {
+  public encounterAnim: EncounterAnim;
+
+  constructor(encounterAnim: EncounterAnim, user: Pokemon, target?: Pokemon) {
+    super(user, target || user);
+
+    this.encounterAnim = encounterAnim;
+  }
+
+  getAnim(): AnimConfig {
+    return encounterAnims.get(this.encounterAnim);
+  }
+
+  isOppAnim(): boolean {
```

Comment:
Usually the word `enemy` is used. Only `overrides` uses `OPP` for some reasons

```suggestion
  isEnemyAnim(): boolean {
```

I'm also not a fan of abbreviating the word `animation` but that is a way bigger change so just thinking out loud

Reply:
unfortunately, this is bad naming leftover from base class `BattleAnim`. Would have to refactor all animation classes to rename

Reply:
don't do it then